systemd/systemd_test.go

```diff
@@ -604,20 +808,43 @@ NeedDaemonReload=no
 }
 
 func (s *SystemdTestSuite) TestStopTimeout(c *C) {
-	restore := MockStopDelays(time.Millisecond, 25*time.Second)
+	// Poll rate = 2ms, Silence timeout = 0ms
+	restore := MockStopDelays(2*time.Millisecond, time.Duration(0))
 	defer restore()
-	err := New(SystemMode, s.rep).Stop([]string{"foo"}, 10*time.Millisecond)
+
+	// 'systemctl stop'
+	s.stopDelays = []time.Duration{
+		// The delay we set here applies to the systemctl stop
+		// command. The purpose is to simulate how long systemd
+		// tries to stop the supplied services before returning.
+		50 * time.Millisecond,
+	}
+
+	err := New(SystemMode, s.rep).Stop([]string{"foo"})
 	c.Assert(err, FitsTypeOf, &Timeout{})
-	c.Assert(len(s.rep.msgs) > 0, Equals, true)
+	c.Assert(len(s.rep.msgs), Equals, 1)
+	// Once the notification silence expires, expect a list of still waiting
+	// services. In this test no mock output is supplied, which results in
+	// systemctl returning nothing, which defaults to running.
+	// See the 'myRun(...)' mock systemctl implementation and 'isStopDone' regex.
 	c.Check(s.rep.msgs[0], Equals, `Waiting for "foo" to stop.`)
 }
 
 func (s *SystemdTestSuite) TestStopManyTimeout(c *C) {
```

Comment:
same question

Reply:
Same explanation applies, but testing with more than one unit, and verifying the notification.